kms/key/Acornfile

```diff
@@ -0,0 +1,135 @@
+	keySpec: "SYMMETRIC_DEFAULT"
+	// The usage for the Key. Options are "ENCRYPT_DECRYPT", "SIGN_VERIFY", and "GENERATE_VERIFY_HMAC". Each keySpec is only compatible with certain keyUsages - see README for more info. Default is "ENCRYPT_DECRYPT".
+	keyUsage: "ENCRYPT_DECRYPT"
+	// The time (in days) that must pass after key deletion is requested before the key is deleted. Default is 30. Minimum is 7. Maximum is 30.
+	pendingWindowDays: 30
+	// AWS IAM policy to attach to the Key. Optional.
+	keyPolicy: {}
+}
+
+services: key: {
+	name: "AWS KMS Key"
+	generated: job: "apply"
+	consumer: permissions: rules: [{
+		apiGroups: ["aws.acorn.io"]
+		verbs: [
+			"kms:Decrypt",
+			"kms:DescribeKey",
+			"kms:Encrypt",
+			"kms:GenerateDataKey",
+			"kms:GenerateDataKeyPair",
+			"kms:GenerateDataKeyPairWithoutPlaintext",
+			"kms:GenerateMac",
+			"kms:GenerateRandom",
+			"kms:GetKeyPolicy",
+			"kms:GetKeyRotationStatus",
+			"kms:GetPublicKey",
+			"kms:ListAliases",
+			"kms:ListGrants",
+			"kms:ListKeyPolicies",
+			"kms:ListResourceTags",
+			"kms:ListRetirableGrants",
+			"kms:ReEncryptFrom",
+			"kms:ReEncryptTo",
+			"kms:Sign",
+			"kms:Verify",
+			"kms:VerifyMac",
+		]
+		resources: ["*"]
+	}]
+}
+
+jobs: apply: {
+	build: {
+		context: "."
+		additionalContexts: {
+			common: "../../libs"
+			utils:  "../../utils"
+		}
+	}
+	files: "/app/config.json": std.toJSON(args)
+	env: {
+		CDK_DEFAULT_ACCOUNT: "@{secrets.aws-context.account-id}"
+		CDK_DEFAULT_REGION:  "@{secrets.aws-context.aws-region}"
+		VPC_ID:              "@{secrets.aws-context.vpc-id}"
+		ACORN_ACCOUNT:       "@{acorn.account}"
+		ACORN_NAME:          "@{acorn.name}"
+		ACORN_PROJECT:       "@{acorn.project}"
+		ACORN_EXTERNAL_ID:   "@{acorn.externalId}"
+	}
+	events: ["create", "update", "delete"]
+	permissions: rules: [{
+		apiGroup: "aws.acorn.io"
+		verbs: [
+			"cloudformation:DescribeStacks",
+			"cloudformation:CreateChangeSet",
+			"cloudformation:DescribeChangeSet",
+			"cloudformation:DescribeStackEvents",
+			"cloudformation:DescribeStackResources",
+			"cloudformation:ExecuteChangeSet",
+			"cloudformation:PreviewStackUpdate",
+			"cloudformation:UpdateStack",
+			"cloudformation:GetTemplateSummary",
+			"cloudformation:DeleteStack",
+			"kms:*",
+		]
+		resources: ["*"]
+	}, {
+		apiGroup: "aws.acorn.io"
+		verbs: [
+			"ec2:DescribeAvailabilityZones",
+			"ec2:DescribeVpcs",
+			"ec2:DescribeSubnets",
+			"ec2:DescribeRouteTables",
+		]
+		resources: ["*"]
+	}, {
+		apiGroup: "api.acorn.io"
+		verbs: [
+			"create",
+		]
+		resources: ["events"]
+	}]
+}
+
+secrets: "aws-context": {
+	name:     "AWS Context"
+	external: "context://aws"
+	type:     "opaque"
+	data: {
+		"account-id": ""
+		"vpc-id":     ""
+		"aws-region": ""
+	}
+}
+
+localData: info: """
+Deleting this acorn will not delete the KMS key.
```

Comment:
Why? This wont be possible in sandbox, and this one has uses for other service. KMS keys are used in pretty much every service that does encryption at rest.

Reply:
In my initial design discussion we concluded that we wanted to protect the user from accidentally deleting keys. But since they must wait at least 7 days anyway, maybe it makes sense to actually schedule the deletion when the CF stack is deleted, instead of orphaning the key like we currently do. Do you think that would be okay?

Reply:
Yeah, I think we need to remove it. Is there a way to bring the key back after you scheduled the deletion?

Reply:
Yes, in AWS you can go and cancel the key deletion. Obviously sandbox users would not be able to do this, but people using their own accounts could.

With the changes that I made to this service acorn, the user can configure (via the `removalPolicy` arg) whether the key will be orphaned or scheduled for deletion when the service acorn is deleted. Should I force it instead to always schedule deletion? I could also possibly add in some logic to detect whether the acorn is in a sandbox region, and force removal if it is and ignore the user input for that parameter.

@cloudnautique

Reply:
If it can be "saved" once it is scheduled for deletion, then I would suggest we remove the userconfigurable removalpolicy and always set to 7, but allow setting it longer.

For now, hold off on detection of sandbox vs. non. We need to figure out the "pro" acorn story.